connman/plugins/blacklist_monitor.c

```diff
@@ -0,0 +1,610 @@
+/*
+ *  Connection Manager
+ *
+ *  Copyright (C) 2022 Jolla Ltd. All rights reserved.
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License version 2 as
+ *  published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <errno.h>
+
+#define CONNMAN_API_SUBJECT_TO_CHANGE
+#include <connman/device.h>
+#include <connman/ipconfig.h>
+#include <connman/inet.h>
+#include <connman/log.h>
+#include <connman/network.h>
+#include <connman/plugin.h>
+#include <connman/service.h>
+
+#include "src/connman.h"
+
+#define IPV6_ANY "::"
+
+struct rtnl_gw {
+	int index;
+	char *gw;
+	int metric;
+};
+
+static GSList *rtnl_gw_list = NULL;
+static GSList *rtnl_gw_restore_list = NULL;
+
+static struct rtnl_gw *new_rtnl_gw(int index, const char *gw, int metric)
+{
+	struct rtnl_gw *item;
+
+	item = g_try_new0(struct rtnl_gw, 1);
+	if (!item) {
+		connman_error("blacklist_monitor.c:new_rtnl_gw() failed to "
+					"create new struct rtnl_gw");
+		return NULL;
+	}
+
+	item->index = index;
+	item->gw = g_strdup(gw);
+	item->metric = metric;
+
+	return item;
+}
+
+static struct rtnl_gw *copy_rtnl_gw(struct rtnl_gw* item)
+{
+	if (!item)
+		return NULL;
+
+	return new_rtnl_gw(item->index, item->gw, item->metric);
+}
+
+static void free_rtnl_gw(gpointer data)
+{
+	struct rtnl_gw *item = data;
+
+	if (!item)
+		return;
+
+	g_free(item->gw);
+	g_free(item);
+}
+
+static gint compare_route(gconstpointer a, gconstpointer b)
+{
+	const struct rtnl_gw *route_a = a;
+	const struct rtnl_gw *route_b = b;
+
+	if (route_a->index == route_b->index &&
+				!g_strcmp0(route_a->gw, route_b->gw) &&
+				route_a->metric == route_b->metric)
+		return 0;
+
+	return -1;
+}
+
+static gint compare_route_by_index(gconstpointer a, gconstpointer b) {
```

Comment:
Inconsistent use of `{`

Reply:
Yeah, fixed along 3e7c78cfe09a61704f71326ad4192a2e86d9451c